### ISSUE TITLE
Getting `go-header_bad.go:1: missing error "Expected:TITLE\\., Actual: TITLE!"` when running make

### ISSUE THREAD

**Issue by purejgleason** [golangci/golangci-lint#2613]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

I am trying to make the library from source. I clone the repo and run `make` and I get 

```
    linters_test.go:37: 
        	Error Trace:	linters_test.go:37
        	            				linters_test.go:216
        	            				linters_test.go:57
        	Error:      	Received unexpected error:
        	            	go-header_bad.go:1: missing error "Expected:TITLE\\., Actual: TITLE!"
        	Test:       	TestSourcesFromTestdataWithIssuesDir/go-header_bad.go
```

This fails the build. I am running this on an Intel Macbook

### Version of golangci-lint

<details>
Built from source
</details>

### Configuration file

<details>
n/a
</details>

### Go environment

<details>

```console
$ go version && go env
# go version devel go1.18-b37a5391f9 Tue Dec 7 20:14:56 2021 +0000 darwin/amd64
```

</details>

### Verbose output of running

<details>

n/a

</details>

### Code example or link to a public repository

<details>

n/a

</details>

**Comment by ldez** [comment]
Hello,

If you want to build a binary, you can use `make build` instead of `make`.
The target `make` run all the tests, and your logs display a test failure, not a build failure.

The target `make` is called by our CI (linux, macos) for all the PRs and the commits, the tests are working.

Maybe it's just flaky behavior, I will check.

**Comment by ldez** [comment]
I have the same problem locally:

```console
$ T=go-header_bad.go make test_linters
GL_TEST_RUN=1 go test -v ./test -count 1 -run TestSourcesFromTestdataWithIssuesDir/go-header_bad.go
=== RUN   TestSourcesFromTestdataWithIssuesDir
    linters_test.go:41: testdata/*.go
=== RUN   TestSourcesFromTestdataWithIssuesDir/go-header_bad.go
=== PAUSE TestSourcesFromTestdataWithIssuesDir/go-header_bad.go
=== CONT  TestSourcesFromTestdataWithIssuesDir/go-header_bad.go
    linters_test.go:216: [run --allow-parallel-runners --disable-all --print-issued-lines=false --out-format=line-number --max-same-issues=100 -Egoheader -c testdata/configs/go-header.yml testdata/go-header_bad.go]
    linters_test.go:37: 
                Error Trace:    linters_test.go:37
                                                        linters_test.go:217
                                                        linters_test.go:57
                Error:          Received unexpected error:
                                go-header_bad.go:1: missing error "Expected:TITLE\\., Actual: TITLE!"
                Test:           TestSourcesFromTestdataWithIssuesDir/go-header_bad.go
--- FAIL: TestSourcesFromTestdataWithIssuesDir (0.34s)
    --- FAIL: TestSourcesFromTestdataWithIssuesDir/go-header_bad.go (0.60s)
FAIL
FAIL    github.com/golangci/golangci-lint/test  1.352s
FAIL
make: *** [Makefile:43: test_linters] Error 1
```

**Comment by purejgleason** [comment]
Also this seems to be a quick fix. I get....
```

WARN [runner] The linter 'scopelint' is deprecated (since v1.39.0) due to: The repository of the linter has been deprecated by the owner.  Replaced by exportloopref. 
WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive. 
```

Which can be fixed by a quick edit of `.golangci.yml` should I create a PR including that and the lint ignore stuff?

**Comment by purejgleason** [comment]
Also @ldez thanks for the advice, yeah I figured it wasn't that important but still wanted to run tests so that is why I just ignored it. It was deprecated and not removed so for now not a huge issue.

**Comment by purejgleason** [comment]
Oh for some reason it isn't showing the fix. I could fix it with...

    -       subj := certPool.Subjects()
    +       subj := certPool.Subjects() //nolint

**Comment by purejgleason** [comment]
The other 2 are just...

```
-        - golint
+        - revive 

-        - scopelint
+        - exportloopref
```

**Comment by ldez** [comment]
[.golangci.yml](https://github.com/golangci/golangci-lint/blob/master/.golangci.yml) doesn't not contains deprecated linters.

There is a [.golangci.example.yml](https://github.com/golangci/golangci-lint/blob/master/.golangci.example.yml) file with all supported options, their description, and default values. This file is neither a working example nor recommended configuration, it's just a reference to display all the configuration options.

https://golangci-lint.run/usage/configuration/#config-file

**Comment by purejgleason** [comment]
@ldez Sorry got my issues mixed up ignore those previous comments I don't know if I would close it
